### PR TITLE
WT-4384 Add documentation about session statistics cursor.

### PR DESCRIPTION
**Pull request by nehakhatri5**

### CODE REVIEW

**Comment by sulabhM** [comment]
@nehakhatri5 have you checked that the generated docs look as expected ?

Other than the generated docs need to be verified, the changes LGTM

**Comment by nehakhatri5** [comment]
@bvpvamsikrishna I have updated the statistics.docx, please  take a look.
@sulabhM I have checked the updated content in the generated html files, looks as expected.

File: src/docs/statistics.dox [hunk 1]
```diff
@@ -74,10 +74,10 @@ when the cursor was opened), as well as statistics logging, when it is
 configured.
 
 The following example configures WiredTiger to maintain only relatively
-inexpensive statistics, and to clears statistics after they are gathered
-or logged:
+inexpensive statistics, and to clear statistics after they are gathered or
+logged:
 
-@snippet ex_all.c Statistics clear configuration
+@snippet ex_all.c Statistics cursor session
```
Comment (bvpvamsikrishna) [inline src/docs/statistics.dox]: snippet should be "clear configuration" not "cursor session" ?. is my understanding wrong.

File: src/include/wiredtiger.in [hunk 2]
```diff
@@ -361,7 +361,8 @@ struct __wt_cursor {
 	 * with WT_CURSOR::prev will move to the last record.
 	 *
 	 * In the case of a statistics cursor, resetting the cursor refreshes
-	 * the statistics information returned.
+	 * the statistics information returned. Resetting a session statistics
+	 * cursor sets all the session statistics values to zero.
```
Comment (bvpvamsikrishna) [inline src/include/wiredtiger.in]: I think better to use "resets" instead of "sets".

File: src/docs/statistics.dox (hunk 1 above)
Comment (nehakhatri5) [reply src/docs/statistics.dox]: That's right. Thanks.

File: src/docs/statistics.dox [hunk 3]
```diff
@@ -88,6 +88,14 @@ in the join (see WT_SESSION::join); the uri of each reference cursor appears
 as a prefix in the description field returned as a value by the statistics
 cursor.
 
+The following example opens a statistics cursor on an open WiredTiger session:
+
+@snippet ex_all.c Statistics cursor session
+
+As this statistics cursor opened on a session, the statistics gathered are
```
[the remark this answers is not in the document]
Comment (agorrod) [reply src/docs/statistics.dox]: @nehakhatri5 I believe Sue meant the `clear` flag that can be passed to `WT_SESSION::open_cursor` - as documented here:
http://source.wiredtiger.com/develop/struct_w_t___s_e_s_s_i_o_n.html#afb5b4a69c2c5cafe411b2b04fdc1c75d

[the remark this answers is not in the document]
Comment (nehakhatri5) [reply src/docs/statistics.dox]: @agorrod I am also referring the `clear` flag associated with `WT_SESSION::open_cursor`. The documentation about the flag says `The selection must agree with the database statistics configuration specified to wiredtiger_open`. Based on that, I tried to explain that a `WT_SESSION::open_cursor('statistics:session')` call in mongodb can't use the `clear` flag.

[the remark this answers is not in the document]
Comment (nehakhatri5) [reply src/docs/statistics.dox]: @sueloverso Please take a look at the explanation above. If that clarifies please update the PR accordingly.

[the remark this answers is not in the document]
Comment (sueloverso) [reply src/docs/statistics.dox]: Okay. I understand how we got here although it feels a bit clunky. In that case, I think the documentation needs to clarify these potential combinations. What, if anything, does the user need to be aware of if they use `cursor->reset` with or without `clear` on `open_cursor`?

**Pull request merged**